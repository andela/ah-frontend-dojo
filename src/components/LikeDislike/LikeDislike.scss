// like-dislike styles
* {
  margin: 0;
  padding: 0;
}

#like-dislike {
  height: 4rem;
  width: auto;
}

img.like-dislike-img {
  width: 1.5rem;
  height: 1.5rem;
  float: left;
  margin: 0.2rem;
}

img#dislike-img {
  width: 1.5rem;
  height: 1.5rem;
  transform-origin: center;
  transform: rotate(180deg);
  position: relative;
}

img.dislike-activated, img.like-activated {
  opacity: 1.0;
}

img.like-dislike-deactivated {
  opacity: 0.5;
}

.like-dislike-wrapper {
  float: right;
}

button {
  background-color: none;
  cursor: pointer ;
  background: none;
  box-sizing: border-box;
  margin: 0;
  border-width: 0;
  float: left;
}

button:focus {
  outline: none;
}