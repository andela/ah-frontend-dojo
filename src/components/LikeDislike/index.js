// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// styles
import './LikeDislike.scss';
/**
 * This is a function which return the LikeDislike component
 * @param {*} n - mixed parameter types*
 * @return {jsx}
 * @example
 * foo = (props) => {}
 */
const LikeDislike = (props) => {
  const {
    dislikeState,
    likeState,
    likeFunction,
    dislikeFunction
  } = props;

  /**
* This is a function which adds a class to the like image button
* depending on whether its state is true or false.
* @param {none}
* @return {string}
* @example
* foo = () => {string}
*/
  const isLiked = () => {
    if (likeState) {
      return 'like-activated';
    }
    return 'like-dislike-deactivated';
  };

  /**
 * This is a function which adds a class to the dislike image button
 * depending on whether its state is true or false.
 * @param {none}
 * @return {string}
 * @example
 * foo = () => {string}
 */
  const isDisLiked = () => {
    if (dislikeState) {
      return 'dislike-activated';
    }
    return 'like-dislike-deactivated';
  };

  return (
    <div id="like-dislike-buttons-wrapper">
      <div className="like-dislike-wrapper">
        <button onClick={dislikeFunction} type="button"><img className={`like-dislike-img  ${isDisLiked()}`} id="dislike-img" src="src/assets/images/like.png" alt="like" /></button>
      </div>
      <div className="like-dislike-wrapper">
        <button onClick={likeFunction} type="button"><img className={`like-dislike-img  ${isLiked()}`} id="like-img" src="src/assets/images/like.png" alt="dislike" /></button>
      </div>
    </div>
  );
};

// props validations
LikeDislike.propTypes = {
  likeState: PropTypes.bool,
  dislikeState: PropTypes.bool,
  likeFunction: PropTypes.func,
  dislikeFunction: PropTypes.func,
};

// default props
LikeDislike.defaultProps = {
  likeState: true,
  dislikeState: false,
  likeFunction: () => { },
  dislikeFunction: () => { },
};

export default LikeDislike;
