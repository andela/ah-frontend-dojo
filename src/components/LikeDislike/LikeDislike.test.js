// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import LikeDislike from '.';

// props
const props = {
  dislikeState: false,
  likeState: false,
};

describe('tests for fucntionality of LikeDislike Component', () => {
  const component = shallow(<LikeDislike {...props} />);
  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('activates like when disliked an article', () => {
    component.setProps({ likeState: true });
    expect(component.find('.like-activated').exists()).toBe(true);
  });

  it('activates dislike when disliked an article', () => {
    component.setProps({ dislikeState: true });
    expect(component.find('.dislike-activated').exists()).toBe(true);
  });
});
