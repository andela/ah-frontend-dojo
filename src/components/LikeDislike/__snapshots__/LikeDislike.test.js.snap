// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for fucntionality of LikeDislike Component matches the snapshot 1`] = `
<div
  id="like-dislike-buttons-wrapper"
>
  <div
    className="like-dislike-wrapper"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt="like"
        className="like-dislike-img  like-dislike-deactivated"
        id="dislike-img"
        src="src/assets/images/like.png"
      />
    </button>
  </div>
  <div
    className="like-dislike-wrapper"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt="dislike"
        className="like-dislike-img  like-dislike-deactivated"
        id="like-img"
        src="src/assets/images/like.png"
      />
    </button>
  </div>
</div>
`;
